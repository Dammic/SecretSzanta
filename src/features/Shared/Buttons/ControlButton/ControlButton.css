@import 'variables/variables.css';

.control {
    padding: 4px 12px;
    background-color: var(--button-control-bg);
    color: var(--button-control-font-color);

    &:hover {
        background-color: var(--button-control-bg-hover);
    }
}
