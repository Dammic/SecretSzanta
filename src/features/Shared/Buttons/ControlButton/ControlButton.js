import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { Button } from '..'

import styles from './ControlButton.css'

const ControlButton = (props) => {
    const { className, children } = props
    return (
        <Button
            {...props}
            className={classNames(styles.control, className)}
        >
            <span className={styles.buttonText}>{children}</span>
        </Button>
    )
}

ControlButton.displayName = 'ControlButton'
ControlButton.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
}

export default ControlButton
