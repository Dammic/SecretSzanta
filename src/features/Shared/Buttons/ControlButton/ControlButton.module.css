@import 'variables/variables.css';

.control {
    padding: 4px 12px;
    border-radius: 4px;
    background-color: var(--button-control-bg);
    color: var(--button-control-font-color);
    text-shadow: 1px 4px 6px #def, 0 0 0 #000, 1px 4px 6px #def;
    font-weight: bold;
    border: 0;
    box-shadow: 0px 0px 2px 0px rgba(0,0,0,0.75);

    &:hover {
        background-color: var(--button-control-bg-hover);
        color: var(--button-control-font-color-hover);
    }
}

.buttonText:before {
    content: "â€¢";
    margin-right: 2px;
    position: relative;
    bottom: 1px;
}
