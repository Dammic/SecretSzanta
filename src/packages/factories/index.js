export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
