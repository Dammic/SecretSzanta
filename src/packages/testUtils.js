import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
