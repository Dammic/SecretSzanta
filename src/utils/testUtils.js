import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
