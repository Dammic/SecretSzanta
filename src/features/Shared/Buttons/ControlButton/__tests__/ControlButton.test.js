import React from 'react'
import { shallow } from 'enzyme'
import { expectMatchingSnapshot } from 'packages/testUtils'
import ControlButton from '../ControlButton'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        children: <span>mockLabel</span>,
        className: 'mockClassName',
        ...propsOverrides,
    }
    const component = renderMethod(<ControlButton {...props} />)
    return { props, component }
}

describe('<ControlButton />', () => {
    it('matches snapshot', () => {
        const { props } = setupProps()
        expectMatchingSnapshot(<ControlButton {...props} />)
    })
})
