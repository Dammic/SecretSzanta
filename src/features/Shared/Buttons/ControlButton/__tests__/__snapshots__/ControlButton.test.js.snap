// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlButton /> matches snapshot 1`] = `
<div
  className="btn control mockClassName"
  role="button"
  tabIndex="0"
>
  <span
    className="buttonText"
  >
    <span>
      mockLabel
    </span>
  </span>
</div>
`;
